### PR TITLE
Add ARB to Arbitrum V3 Pool - Review

### PR DESCRIPTION
**Pull request by marczeller**

### CODE REVIEW

File: src/AaveV3_Arb_ARBListing_20231207/ARBListing.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+By enabling borrowing, it allows users and institutions to take a unique position on this Governance token as the network matures and builds a decentralized community.
+
+Furthermore, it more closely aligns the Aave community with Arbitrum and, by extension, Ethereum.
+
+Additionally, it allows for creative uses of ARB received by the DAO to seed initial liquidity.
+
+## Specification
+
+Ticker: RPL
```
Comment (sakulstra) [inline src/AaveV3_Arb_ARBListing_20231207/ARBListing.md:23]:
```suggestion
Ticker: ARB
```

File: src/AaveV3_Arb_ARBListing_20231207/ARBListing.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+| Uoptimal                 | 45%      |
+| Stable Borrowing         | Disabled |
+
+## References
+
+- Implementation: [Arbitrum](https://github.com/bgd-labs/aave-proposals/blob/main/src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol)
+- Tests: [Arbitrum](https://github.com/bgd-labs/aave-proposals/blob/main/src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.t.sol)
+- [Snapshot](https://snapshot.org/#/aave.eth/proposal/0x308439517ffc8faa8709db0b4a1d131d2402ee8a3282cb79adf890de6135ec98)
+- [Discussion](https://governance.aave.com/t/arfc-add-arb-to-arbitrum-aave-v3/13225)
+
```
Comment (sakulstra) [inline src/AaveV3_Arb_ARBListing_20231207/ARBListing.md:54]: perhaps also link previous discussion, as from the one linked it's not clear what gauntlet/chaos said https://governance.aave.com/t/temp-check-add-arb-to-arbitrum-aave-v3/12915

File: src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  {
+    IEngine.ListingWithCustomImpl[] memory listings = new IEngine.ListingWithCustomImpl[](1);
+
+    listings[0] = IEngine.ListingWithCustomImpl(
+      IEngine.Listing({
+      asset: ARB,
+      assetSymbol: 'ARB',
+      priceFeed: ARB_USD_FEED,
+      rateStrategyParams: Rates.RateStrategyParams({
+        optimalUsageRatio: _bpsToRay(80_00),
```
Comment (sakulstra) [inline src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol:32]: should be 45 according to recommendation

Comment (marczeller) [reply src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol]: good catch pushed.

File: src/AaveV3_Arb_ARBListing_20231207/ARBListing.md (hunk 2 above)
Comment (marczeller) [reply src/AaveV3_Arb_ARBListing_20231207/ARBListing.md:55]: fair point, added TEMP CHECK link

File: src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      stableRateModeEnabled: EngineFlags.DISABLED,
+      borrowableInIsolation: EngineFlags.DISABLED,
+      withSiloedBorrowing: EngineFlags.DISABLED,
+      flashloanable: EngineFlags.ENABLED,
+      ltv: 50_00,
+      liqThreshold: 60_00,
+      liqBonus: 10_00,
+      reserveFactor: 20_00,
+      supplyCap: 20_000_000,
+      borrowCap: 16_000_000,
```
Comment (brotherlymite) [inline src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol:52]: The borrow cap on the AIP and here are different (16.5M on AIP text and 16M here)

File: src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        variableRateSlope2: _bpsToRay(300_00),
+        stableRateSlope1: _bpsToRay(7_00),
+        stableRateSlope2: _bpsToRay(300_00),
+        baseStableRateOffset: _bpsToRay(1_00),
+        stableRateExcessOffset: _bpsToRay(8_00),
+        optimalStableToTotalDebtRatio: _bpsToRay(20_00)
+      }),
+      enabledToBorrow: EngineFlags.ENABLED,
+      stableRateModeEnabled: EngineFlags.DISABLED,
+      borrowableInIsolation: EngineFlags.DISABLED,
```
Comment (brotherlymite) [inline src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol:44]: `borrowableInIsolation` should be enabled no?

File: src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol [hunk 6]
```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {AaveV3PayloadArbitrum, IEngine, Rates, EngineFlags} from 'aave-helpers/v3-config-engine/AaveV3PayloadArbitrum.sol';
+import {AaveV3Arbitrum, AaveV3ArbitrumAssets} from 'aave-address-book/AaveV3Arbitrum.sol';
```
Comment (brotherlymite) [inline src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol:5]:
```suggestion
import {AaveV3Arbitrum} from 'aave-address-book/AaveV3Arbitrum.sol';
```

File: src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.t.sol [hunk 7]
```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import 'forge-std/Test.sol';
```
Comment (brotherlymite) [inline src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.t.sol:4]:
```suggestion
```

File: src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.t.sol [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  address public constant ARB = 0x912CE59144191C1204E64559FE8253a0e49E6548;
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('arbitrum'), 110366299);
+  }
+
+  function testProposalExecution() public {
+    AaveV3_Arb_ARBListing_20231207 proposal = new AaveV3_Arb_ARBListing_20231207();
+
+    ReserveConfig[] memory allConfigsBefore = createConfigurationSnapshot(
```
Comment (brotherlymite) [inline src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.t.sol:26]:
```suggestion
    createConfigurationSnapshot(
```

File: src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol (hunk 5 above)
Comment (brotherlymite) [reply src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol:44]: nvm it should be disabled

File: src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol (hunk 4 above)
Comment (marczeller) [reply src/AaveV3_Arb_ARBListing_20231207/AaveV3_Arb_ARBListing_20231207.sol]: good catch, fixed.

File: diffs/preAaveV3_Arb_ARBListing_20231207_postAaveV3_Arb_ARBListing_20231207.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+#### ARB ([0x912CE59144191C1204E64559FE8253a0e49E6548](https://arbiscan.io/address/0x912CE59144191C1204E64559FE8253a0e49E6548))
+
+| description | value |
+| --- | --- |
+| decimals | 18 |
+| isActive | true |
+| isFrozen | false |
+| supplyCap | 20,000,000 ARB |
+| borrowCap | 16,000,000 ARB |
```
Comment (brotherlymite) [inline diffs/preAaveV3_Arb_ARBListing_20231207_postAaveV3_Arb_ARBListing_20231207.md:13]: Can you also update the diffs as they have changed :)

Comment (marczeller) [reply diffs/preAaveV3_Arb_ARBListing_20231207_postAaveV3_Arb_ARBListing_20231207.md]: fixed

**Review by sakulstra — APPROVED** [review]

**Pull request merged**